lib/adapters/websockex.ex

```diff
@@ -31,17 +37,17 @@ if Code.ensure_loaded?(WebSockex) do
       end
     end
 
-    @impl true
+    @impl Adapter
     def disconnect(client) do
       Kernel.send(client, :close)
     end
 
     defp start_link(url, timeout, args) do
-      websockex_opts = [
-        extra_headers: [{"Sec-WebSocket-Protocol", "janus-protocol"}]
-      ]
+      %{extra_headers: extra_headers} = args
+
+      opts = [extra_headers: extra_headers]
 
-      case WebSockex.start_link(url, __MODULE__, args, websockex_opts) do
+      case WebSockex.start_link(url, __MODULE__, args, opts) do
```

Comment:
As mentioned, extract the implementation of Websockex callback module (can be in the same file) to have clear separation between 2 processes ( now we have callbacks executed in different context in one module)